src/lib/libast/string/fmtre.c

```diff
@@ -66,20 +67,24 @@ char *fmtre(const char *as) {
     } else {
         s++;
     }
-    while (1) {
+
+    while (*s) {
         c = *s++;
+        if (c == '*' && !*s) {
+            end = false;
+            break;
```

Comment:
Was not this statement breaking out of `switch` statement ?

Reply:
Yes, but placed in the switch it relied on the `break;` at the bottom of the loop to break out of the loop. It also hid the fact that this was a special-case. Similar to how the `case 0: break;` obscured the fact that the loop is terminated when the end of the string is reached. Which is why I changed `while (1)` to `while (*s)` and dropped that case.